libphx/src/line_segment.rs

```diff
@@ -9,26 +9,22 @@ pub struct LineSegment {
     pub p1: Vec3,
 }
 
+#[luajit_ffi_gen::luajit_ffi(meta = true, clone = true)]
 impl LineSegment {
-    pub fn to_string(&self) -> String {
-        format!("p0:{} p1:{}", self.p0.to_string(), self.p1.to_string(),)
+    pub fn to_ray(&self, out: &mut Ray) {
+        out.p = self.p0;
+        out.dir = self.p1 - self.p0;
+        out.tMin = 0.0f32;
+        out.tMax = 1.0f32;
     }
-}
-
-#[no_mangle]
-pub extern "C" fn LineSegment_ToRay(this: &LineSegment, out: &mut Ray) {
-    out.p = this.p0;
-    out.dir = this.p1 - this.p0;
-    out.tMin = 0.0f32;
-    out.tMax = 1.0f32;
-}
 
-#[no_mangle]
-pub extern "C" fn LineSegment_FromRay(ray: &Ray, out: &mut LineSegment) {
-    Ray_ToLineSegment(ray, out);
-}
+    #[bind(role = "constructor")]
```

Comment:
I'm curious, can we derive the fact that this is a constructor based on the fact it's not got `&self` as the first argument?

Reply:
There are several cases when there is no **self** but it's not a constructor, i.e. **Bit**

Reply:
Probably we can check that method returns **Self**...

Reply:
I guess in the `Bit` example, all the methods are static, so we don't bother generating a metatype section at all. However, for some impls, we're going to have a mixture of static methods and non-static methods. The static methods are the ones that we don't want to put in the metatype section. I'm just thinking that your documentation says:
```
**constructor** - function won't appear in the metatype section
```
which to me, sounds like the word "constructor" should actually be "static_method", and that can be deduced by the lack of `self`.

Do we even treat constructors in a special way other than the fact their static methods?

Reply:
I wrote some more thoughts on discord.

Reply:
Refactored